test/testsuite/wasi_threads_exit_main_block.wat

```diff
@@ -0,0 +1,39 @@
+(module
+  (memory (import "foo" "bar") 1 1 shared)
```

Comment:
Should there also be an exported memory? From the [doc](https://github.com/WebAssembly/WASI/blob/main/legacy/application-abi.md):

> Regardless of the kind, all modules accessing WASI APIs also export a linear memory with the name memory.

Reply:
> Should there also be an exported memory? From the [doc](https://github.com/WebAssembly/WASI/blob/main/legacy/application-abi.md):
> 
> > Regardless of the kind, all modules accessing WASI APIs also export a linear memory with the name memory.

sure. i will update tomorrow.

Reply:
I'm not sure about this.  Firstly, that doc marked as legacy.  Secondly, if this doc were current I would recommend that it be updated as part of wasi-threads to specify that memory but *either* be exported *or* imported.   

IIUC wasi-threads should require that memory be imported.  The extra requirement to redundantly re-export the memory back to the runtime seems like a wasmtime specific thing that I'm not sure we want to spec.

Reply:
i guess the original motivation of memory export in wasi was to explicitly allow the host to examine the linear memory for given pointers, especially for polyfilling implementations. in that regard, you are right when the memory is imported from the host, there is little point to export it back.

on the other hand, it might still make sense to have a well-known name (ie. "memory") to indicate which memory is used for wasi pointers in case of multi-memory.

i personally has no strong opinions either ways.

Reply:
> > Should there also be an exported memory? From the [doc](https://github.com/WebAssembly/WASI/blob/main/legacy/application-abi.md):
> > > Regardless of the kind, all modules accessing WASI APIs also export a linear memory with the name memory.
> 
> sure. i will update tomorrow.

postponed as it seems a bit more controversial than i thought.

Reply:
an old discussion: https://github.com/WebAssembly/WASI/pull/48#discussion_r289338186

Reply:
i'd suggest not to block this PR for the memory export/import issue.
it isn't something to decide in this PR.
https://github.com/WebAssembly/wasi-threads/issues/22

Reply:
> > > Should there also be an exported memory? From the [doc](https://github.com/WebAssembly/WASI/blob/main/legacy/application-abi.md):
> > > > Regardless of the kind, all modules accessing WASI APIs also export a linear memory with the name memory.
> > 
> > 
> > sure. i will update tomorrow.
> 
> postponed as it seems a bit more controversial than i thought.

i added export as well to increase the chances for these to work on more runtimes.